### DISCUSSION TITLE
backtracking_linesearch function

### DISCUSSION THREAD

**Discussion by singhvasu1626** [slimgroup/JUDI.jl#62] [Q&A] [answered]
every time i run the codes on Julia DeVito its throws the error that backtracking_linesearch function is not defined.
also in calculating through stochastic gradient it tells "m" not defined.
I am not able to resolve why it is giving this error. please help me as soon as possible!!!

**Comment by mloubout** [comment, marked as answer]
Hi, can I ask what version of JUDI and of the examples you are running?

**Reply by singhvasu1626** [reply]
thanks for replying
sir i am running examples given in the leading edge tutorial>>fwi and algorithms and I have installed JUDI as instructed Pkg.clone("https://github.com/slimgroup/JUDI.jl.git") by typing this so may be it gives me the latest version
also I am using JULIA v1.5.4

**Reply by mloubout** [reply]
The leading edge tutorial correspond to an older version of JUDI (from 2019). You can find the updated corresponding examples for the newest JUDI version as

https://github.com/slimgroup/JUDI.jl/blob/master/examples/scripts/fwi_example_2D.jl

**Reply by singhvasu1626** [reply]
sir i tried this also...in line
# Projection operator for bound constraints
proj(x) = reshape(median([vec(mmin) vec(x) vec(mmax)]; dims=2),model0.n)
ls = BackTracking(order=3, iterations=10, )
it was giving same error for backtracking that function not defined
and in this line 
  # Update model and bound projection
    model0.m .= proj(model0.m .+ step .* p)
telling "m " not defined

**Reply by mloubout** [reply]
You need to import the proper package as in the example (SlimOptim or LineSearches) for BackTracking to be defined. The script I linked above is running (with one change on line 14 `v0 = sqrt.(1f0 ./ m0)`), so make sure you are `using` all the necessary packages

**Reply by singhvasu1626** [reply]
ohk sir ....i will try again and let you know...
thank you!!

**Reply by singhvasu1626** [reply]
@mloubout thank you sir very much ...now the codes are executing successfully !!!!!
Given a good platform for students like us !!! Excellent work

**Comment by mloubout** [comment]
Great to hear
